### ISSUE TITLE
IPPO with common reward in Cleanup reaches ~2500 reward (more or less comparable to MAPPO)

### ISSUE THREAD

**Issue by jselvaraaj** [cooperativex/SocialJax#6]
For IPPO with a common reward setting in the Cleanup task, I am able to get average episode reward of ~2500, making its performance comparable to MAPPO.

<img width="1738" height="1126" alt="Image" src="https://github.com/user-attachments/assets/418667e5-6e3a-40b0-b96a-13bed7e60227" />

### MAPPO:
```
{
│   'logger': {
│   │   'base_exp_path': 'results/',
│   │   'loggers': {
│   │   │   'console': {
│   │   │   │   '_target_': 'contract_discovery.mava.utils.logger.ConsoleLogger',
│   │   │   │   'enabled': True
│   │   │   },
│   │   │   'wandb': {
│   │   │   │   '_target_': 'contract_discovery.mava.utils.logger.WandbLogger',
│   │   │   │   'enabled': True,
│   │   │   │   'project': 'contract_discovery',
│   │   │   │   'entity': 'marl',
│   │   │   │   'group': 'cleanup_socialjax',
│   │   │   │   'sync_tensorboard': True,
│   │   │   │   'upload_json_data': True
│   │   │   },
│   │   │   'tensorboard': {
│   │   │   │   '_target_': 'contract_discovery.mava.utils.logger.TensorboardLogger',
│   │   │   │   'enabled': True
│   │   │   },
│   │   │   'json': {
│   │   │   │   '_target_': 'contract_discovery.mava.utils.logger.JsonLogger',
│   │   │   │   'enabled': True,
│   │   │   │   'path': None,
│   │   │   │   'task_name': 'cleanup_socialjax',
│   │   │   │   'env_name': 'clean_up',
│   │   │   │   'seed': 42
│   │   │   }
│   │   },
│   │   'checkpointing': {
│   │   │   'save_model': False,
│   │   │   'save_args': {
│   │   │   │   'save_interval_steps': 1000000,
│   │   │   │   'max_to_keep': 1,
│   │   │   │   'keep_period': None,
│   │   │   │   'checkpoint_uid': None,
│   │   │   │   'rel_dir': 'checkpoints'
│   │   │   },
│   │   │   'load_model': False,
│   │   │   'load_args': {'checkpoint_uid': '', 'rel_dir': 'checkpoints'}
│   │   },
│   │   'system_name': 'rec_mappo'
│   },
│   'arch': {
│   │   'architecture_name': 'anakin',
│   │   'num_envs': 16,
│   │   'evaluation_greedy': False,
│   │   'num_evaluation': 50,
│   │   'num_eval_episodes': 2,
│   │   'num_absolute_metric_eval_episodes': 2,
│   │   'absolute_metric': True,
│   │   'render_every_n_eval': 48,
│   │   'num_render_episodes': 1,
│   │   'num_render_steps': 50
│   },
│   'system': {
│   │   'total_timesteps': 16384000,
│   │   'num_updates': 1000,
│   │   'seed': 42,
│   │   'add_agent_id': 0,
│   │   'actor_lr': 0.001,
│   │   'critic_lr': 0.001,
│   │   'update_batch_size': 2,
│   │   'rollout_length': 128,
│   │   'ppo_epochs': 4,
│   │   'num_minibatches': 2,
│   │   'gamma': 0.99,
│   │   'gae_lambda': 0.95,
│   │   'clip_eps': 0.2,
│   │   'ent_coef': 0.01,
│   │   'vf_coef': 0.5,
│   │   'max_grad_norm': 0.5,
│   │   'decay_learning_rates': True,
│   │   'recurrent_chunk_size': 128,
│   │   'num_agents': 8,
│   │   'num_updates_per_eval': 20
│   },
│   'network': {
│   │   'hidden_state_dim': 64,
│   │   'actor_network': {
│   │   │   'pre_torso': {
│   │   │   │   '_target_': 'contract_discovery.mava.networks.torsos.CNNTorso',
│   │   │   │   'channel_sizes': [32, 32],
│   │   │   │   'kernel_sizes': [3, 3],
│   │   │   │   'strides': [1, 1],
│   │   │   │   'use_layer_norm': False,
│   │   │   │   'activation': 'relu'
│   │   │   },
│   │   │   'post_torso': {
│   │   │   │   '_target_': 'contract_discovery.mava.networks.torsos.MLPTorso',
│   │   │   │   'layer_sizes': [64],
│   │   │   │   'use_layer_norm': False,
│   │   │   │   'activation': 'relu'
│   │   │   }
│   │   },
│   │   'critic_network': {
│   │   │   'pre_torso': {
│   │   │   │   '_target_': 'contract_discovery.mava.networks.torsos.CNNTorso',
│   │   │   │   'channel_sizes': [32, 32],
│   │   │   │   'kernel_sizes': [3, 3],
│   │   │   │   'strides': [1, 1],
│   │   │   │   'use_layer_norm': False,
│   │   │   │   'activation': 'relu'
│   │   │   },
│   │   │   'post_torso': {
│   │   │   │   '_target_': 'contract_discovery.mava.networks.torsos.MLPTorso',
│   │   │   │   'layer_sizes': [64],
│   │   │   │   'use_layer_norm': False,
│   │   │   │   'activation': 'relu'
│   │   │   }
│   │   }
│   },
│   'env': {
│   │   'env_name': 'clean_up',
│   │   'eval_metric': 'episode_return',
│   │   'implicit_agent_id': False,
│   │   'log_win_rate': False,
│   │   'kwargs': {'use_full_grid_as_global_state': False},
│   │   'scenario': {
│   │   │   'name': 'Cleanup',
│   │   │   'task_name': 'cleanup_socialjax',
│   │   │   'task_config': {
│   │   │   │   'num_agents': 8,
│   │   │   │   'maxAppleGrowthRate': 0.05,
│   │   │   │   'obs_size': 12,
│   │   │   │   'num_inner_steps': 1000,
│   │   │   │   'num_outer_steps': 1,
│   │   │   │   'shared_rewards': True,
│   │   │   │   'inequity_aversion': False,
│   │   │   │   'inequity_aversion_target_agents': [0, 1],
│   │   │   │   'inequity_aversion_alpha': 5,
│   │   │   │   'inequity_aversion_beta': 0.05,
│   │   │   │   'enable_smooth_rewards': False,
│   │   │   │   'svo': False,
│   │   │   │   'svo_target_agents': [0, 1],
│   │   │   │   'svo_w': 0.5,
│   │   │   │   'svo_ideal_angle_degrees': 45
│   │   │   }
│   │   }
│   }
}
```

### IPPO

```
{
│   'logger': {
│   │   'base_exp_path': 'results/',
│   │   'loggers': {
│   │   │   'console': {
│   │   │   │   '_target_': 'contract_discovery.mava.utils.logger.ConsoleLogger',
│   │   │   │   'enabled': True
│   │   │   },
│   │   │   'wandb': {
│   │   │   │   '_target_': 'contract_discovery.mava.utils.logger.WandbLogger',
│   │   │   │   'enabled': True,
│   │   │   │   'project': 'contract_discovery',
│   │   │   │   'entity': 'marl',
│   │   │   │   'group': 'cleanup_socialjax',
│   │   │   │   'sync_tensorboard': True,
│   │   │   │   'upload_json_data': True
│   │   │   },
│   │   │   'tensorboard': {
│   │   │   │   '_target_': 'contract_discovery.mava.utils.logger.TensorboardLogger',
│   │   │   │   'enabled': True
│   │   │   },
│   │   │   'json': {
│   │   │   │   '_target_': 'contract_discovery.mava.utils.logger.JsonLogger',
│   │   │   │   'enabled': True,
│   │   │   │   'path': None,
│   │   │   │   'task_name': 'cleanup_socialjax',
│   │   │   │   'env_name': 'clean_up',
│   │   │   │   'seed': 42
│   │   │   }
│   │   },
│   │   'checkpointing': {
│   │   │   'save_model': False,
│   │   │   'save_args': {
│   │   │   │   'save_interval_steps': 1000000,
│   │   │   │   'max_to_keep': 1,
│   │   │   │   'keep_period': None,
│   │   │   │   'checkpoint_uid': None,
│   │   │   │   'rel_dir': 'checkpoints'
│   │   │   },
│   │   │   'load_model': False,
│   │   │   'load_args': {'checkpoint_uid': '', 'rel_dir': 'checkpoints'}
│   │   },
│   │   'system_name': 'rec_ippo'
│   },
│   'arch': {
│   │   'architecture_name': 'anakin',
│   │   'num_envs': 16,
│   │   'evaluation_greedy': False,
│   │   'num_evaluation': 50,
│   │   'num_eval_episodes': 2,
│   │   'num_absolute_metric_eval_episodes': 2,
│   │   'absolute_metric': True,
│   │   'render_every_n_eval': 0,
│   │   'num_render_episodes': 1,
│   │   'num_render_steps': 50
│   },
│   'system': {
│   │   'total_timesteps': 16384000,
│   │   'num_updates': 1000,
│   │   'seed': 42,
│   │   'add_agent_id': 0,
│   │   'actor_lr': 0.001,
│   │   'critic_lr': 0.001,
│   │   'update_batch_size': 2,
│   │   'rollout_length': 128,
│   │   'ppo_epochs': 4,
│   │   'num_minibatches': 2,
│   │   'gamma': 0.99,
│   │   'gae_lambda': 0.95,
│   │   'clip_eps': 0.2,
│   │   'ent_coef': 0.01,
│   │   'vf_coef': 0.5,
│   │   'max_grad_norm': 0.5,
│   │   'decay_learning_rates': True,
│   │   'recurrent_chunk_size': 128,
│   │   'num_agents': 8,
│   │   'num_updates_per_eval': 20
│   },
│   'network': {
│   │   'hidden_state_dim': 64,
│   │   'actor_network': {
│   │   │   'pre_torso': {
│   │   │   │   '_target_': 'contract_discovery.mava.networks.torsos.CNNTorso',
│   │   │   │   'channel_sizes': [32, 32],
│   │   │   │   'kernel_sizes': [3, 3],
│   │   │   │   'strides': [1, 1],
│   │   │   │   'use_layer_norm': False,
│   │   │   │   'activation': 'relu'
│   │   │   },
│   │   │   'post_torso': {
│   │   │   │   '_target_': 'contract_discovery.mava.networks.torsos.MLPTorso',
│   │   │   │   'layer_sizes': [64],
│   │   │   │   'use_layer_norm': False,
│   │   │   │   'activation': 'relu'
│   │   │   }
│   │   },
│   │   'critic_network': {
│   │   │   'pre_torso': {
│   │   │   │   '_target_': 'contract_discovery.mava.networks.torsos.CNNTorso',
│   │   │   │   'channel_sizes': [32, 32],
│   │   │   │   'kernel_sizes': [3, 3],
│   │   │   │   'strides': [1, 1],
│   │   │   │   'use_layer_norm': False,
│   │   │   │   'activation': 'relu'
│   │   │   },
│   │   │   'post_torso': {
│   │   │   │   '_target_': 'contract_discovery.mava.networks.torsos.MLPTorso',
│   │   │   │   'layer_sizes': [64],
│   │   │   │   'use_layer_norm': False,
│   │   │   │   'activation': 'relu'
│   │   │   }
│   │   }
│   },
│   'env': {
│   │   'env_name': 'clean_up',
│   │   'eval_metric': 'episode_return',
│   │   'implicit_agent_id': False,
│   │   'log_win_rate': False,
│   │   'kwargs': {'use_full_grid_as_global_state': False},
│   │   'scenario': {
│   │   │   'name': 'Cleanup',
│   │   │   'task_name': 'cleanup_socialjax',
│   │   │   'task_config': {
│   │   │   │   'num_agents': 8,
│   │   │   │   'maxAppleGrowthRate': 0.05,
│   │   │   │   'obs_size': 12,
│   │   │   │   'num_inner_steps': 1000,
│   │   │   │   'num_outer_steps': 1,
│   │   │   │   'shared_rewards': True,
│   │   │   │   'inequity_aversion': False,
│   │   │   │   'inequity_aversion_target_agents': [0, 1],
│   │   │   │   'inequity_aversion_alpha': 5,
│   │   │   │   'inequity_aversion_beta': 0.05,
│   │   │   │   'enable_smooth_rewards': False,
│   │   │   │   'svo': False,
│   │   │   │   'svo_target_agents': [0, 1],
│   │   │   │   'svo_w': 0.5,
│   │   │   │   'svo_ideal_angle_degrees': 45
│   │   │   }
│   │   }
│   }
```
